2.6 blog post (#12412)
prettier/prettier: changelog_unreleased/blog-post-intro.md

---
author: "Sosuke Suzuki (@sosukesuzuki)"
authorURL: "https://github.com/sosukesuzuki"
title: "Prettier 2.6: new singleAttributePerLine option and new JavaScript features!"
---

> This release includes a new `singleAttributePerLine` option. For us, adding a new option was a hard decision.

Reviewer: Can you add a few sentences here to describe why the new option is necessary?

Author: Okay, can you review the below sentence?

***

This release includes a new `singleAttributePerLine` option. This is an option to print single attribute per line in HTML, JSX, Vue SFC template. Prettier, following its [Option Philosophy](https://prettier.io/docs/en/option-philosophy.html), does not like to add such an option. However, there are many users who want this feature, and indeed [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript) and [eslint-plugin-vue](https://eslint.vuejs.org/) recommend the singe attribute per line style. A PR ([#6644](https://github.com/prettier/prettier/pull/6644)) to add this feature was opened in October 2019 and has been supported by many until now. For us it was a hard decision to add this option. We hope the addition of this option will benefit many users.

Reviewer: I made a few grammar tweaks (and linked the style guide references to the definition of the rule) but otherwise it looks great!

> This release includes a new `singleAttributePerLine` option. This is an option to print only one attribute per line in Vue SFC templates, HTML, and JSX. Per our [Option Philosophy](https://prettier.io/docs/en/option-philosophy.html), we would prefer not to add such an option. However, there are many users who want this feature, and major style guides like [Airbnb’s JavaScript Style Guide](https://github.com/airbnb/javascript/blob/274c8d570155a05b016980294d4204c5711bce86/packages/eslint-config-airbnb/rules/react.js#L97-L99) and [Vue’s style guide](https://vuejs.org/style-guide/rules-strongly-recommended.html#multi-attribute-elements) recommend the single attribute per line style. A PR (https://github.com/prettier/prettier/pull/6644) to add this feature was opened in October 2019, and both it and the corresponding issue (https://github.com/prettier/prettier/issues/5501) have received a significant amount of support from users. For us it was a hard decision to add this option. We hope the addition of this option will benefit many users without significantly harming our principles.

```suggestion
This release includes a new `singleAttributePerLine` option. This is an option to print only one attribute per line in Vue SFC templates, HTML, and JSX. Per our [Option Philosophy](https://prettier.io/docs/en/option-philosophy.html), we would prefer not to add such an option. However, there are many users who want this feature, and major style guides like [Airbnb’s JavaScript Style Guide](https://github.com/airbnb/javascript/blob/274c8d570155a05b016980294d4204c5711bce86/packages/eslint-config-airbnb/rules/react.js#L97-L99) and [Vue’s style guide](https://vuejs.org/style-guide/rules-strongly-recommended.html#multi-attribute-elements) recommend the single attribute per line style. A PR (https://github.com/prettier/prettier/pull/6644) to add this feature was opened in October 2019, and both it and the corresponding issue (https://github.com/prettier/prettier/issues/5501) have received a significant amount of support from users. For us it was a hard decision to add this option. We hope the addition of this option will benefit many users without significantly harming our principles.
```

Author: thank you!

> We've also support formatting for some new ECMAScript proposals via Babel.

Reviewer: ```suggestion
We've also added support formatting for some new JavaScript syntax proposals via Babel.
```

---
author: "Sosuke Suzuki (@sosukesuzuki)"
authorURL: "https://github.com/sosukesuzuki"
title: "Prettier 2.6: new singleAttributePerLine option and new JavaScript features!"
---

> This release includes a new `singleAttributePerLine` option. This is an option to print only one attribute per line in Vue SFC templates, HTML, and JSX. Per our [Option Philosophy](https://prettier.io/docs/en/option-philosophy.html), we would prefer not to add such an option. However, there are many users who want this feature, and major style guides like [Airbnb’s JavaScript Style Guide](https://github.com/airbnb/javascript/blob/274c8d570155a05b016980294d4204c5711bce86/packages/eslint-config-airbnb/rules/react.js#L97-L99) and [Vue’s style guide](https://vuejs.org/style-guide/rules-strongly-recommended.html#multi-attribute-elements) recommend the single attribute per line style. A PR (https://github.com/prettier/prettier/pull/6644) to add this feature was opened in October 2019, and both it and the corresponding issue (https://github.com/prettier/prettier/issues/5501) have received a significant amount of support from users. For us it was a hard decision to add this option. We hope the addition of this option will benefit many users without significantly harming our principles.

Reviewer 2: ```suggestion
This release includes a new `singleAttributePerLine` option. This is an option to print only one attribute per line in Vue SFC templates, HTML, and JSX. Per our [Option Philosophy](https://prettier.io/docs/en/option-philosophy.html), we would prefer not to add such an option. However, there are many users who want this feature, and major style guides like [Airbnb’s JavaScript Style Guide](https://github.com/airbnb/javascript/blob/274c8d570155a05b016980294d4204c5711bce86/packages/eslint-config-airbnb/rules/react.js#L97-L99) and [Vue’s style guide](https://vuejs.org/style-guide/rules-strongly-recommended.html#multi-attribute-elements) recommend the single attribute per line style. A [PR](https://github.com/prettier/prettier/pull/6644) to add this feature was opened in October 2019, and both it and the [corresponding issue](https://github.com/prettier/prettier/issues/5501) have received a significant amount of support from users. For us it was a hard decision to add this option. We hope the addition of this option will benefit many users without significantly harming our principles.
```
(optional)

The diff is not very clear, so here is what I’ve changed:

```diff
-A PR (https://github.com/prettier/prettier/pull/6644) to add this feature was opened in October 2019, and both it and the corresponding issue (https://github.com/prettier/prettier/issues/5501) have received
+A [PR](https://github.com/prettier/prettier/pull/6644) to add this feature was opened in October 2019, and both it and the [corresponding issue](https://github.com/prettier/prettier/issues/5501) have received
```

We've also added support formatting for some new JavaScript syntax proposals via Babel.

> If you enjoy Prettier and would like to support our work, consider sponsoring us directly via [our OpenCollective](https://opencollective.com/prettier) or by sponsoring the projects we depend on, including [typescript-eslint](https://opencollective.com/typescript-eslint), [remark](https://opencollective.com/unified), and [Babel](https://opencollective.com/babel). Also thanks to [LINE](https://line.me/en/) and [Ubie](https://ubie.life/en) for actually sponsoring us! For more information on LINE's OSS sponsoring, please read the blog post [LINE has started to donate to Prettier – why companies should support OSS](https://engineering.linecorp.com/en/blog/support-prettier-and-oss/).

Reviewer 3: ```suggestion
If you enjoy Prettier and would like to support our work, consider sponsoring us directly via [our OpenCollective](https://opencollective.com/prettier) or by sponsoring the projects we depend on, including [typescript-eslint](https://opencollective.com/typescript-eslint), [remark](https://opencollective.com/unified), and [Babel](https://opencollective.com/babel). Also thanks to [LINE Corporation](https://opencollective.com/line), [Ubie](https://opencollective.com/ubie-inc) and many others for actually sponsoring us! For more information on LINE's OSS sponsoring, please read the blog post [LINE has started to donate to Prettier – why companies should support OSS](https://engineering.linecorp.com/en/blog/support-prettier-and-oss/).
```

Let's link to their opencollective account?

Author: I don't think we should link to their  OpenCollective. This is because the company that sponsored us would want us to see the company's website rather than the OpenCollective link.

Reviewer 3: They have their production links on OpenCollective account.

Anyway, can you add "Airbnb" and "Nx"? They are doing generous donation monthly.

Author: > Anyway, can you add "Airbnb" and "Nx"? They are doing generous donation monthly.

Sure!

Btw, as you mentioned in Discord, I  think we should add sponsors in our website.
